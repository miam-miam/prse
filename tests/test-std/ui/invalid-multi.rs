use prse::parse;

fn main() {
    let l = "test: 5";

    parse!(l, "test: {:}");
    parse!(l, "test: {::}");
    parse!(l, "test: {:,:999}");
    parse!(l, "test: {:,:-1}");
    parse!(l, "test: {:,:,}");
}
