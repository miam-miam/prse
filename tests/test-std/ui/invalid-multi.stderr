error: invalid multi parse, it must be of the form <var>:<sep>:<count>.
 --> ui/invalid-multi.rs:6:15
  |
6 |     parse!(l, "test: {:}");
  |               ^^^^^^^^^^^

error: expected a number between 0 and 255 but found 999.
 --> ui/invalid-multi.rs:7:15
  |
7 |     parse!(l, "test: {:,:999}");
  |               ^^^^^^^^^^^^^^^^

error: expected a number between 0 and 255 but found -1.
 --> ui/invalid-multi.rs:8:15
  |
8 |     parse!(l, "test: {:,:-1}");
  |               ^^^^^^^^^^^^^^^

error: expected a number between 0 and 255 but found ,.
 --> ui/invalid-multi.rs:9:15
  |
9 |     parse!(l, "test: {:,:,}");
  |               ^^^^^^^^^^^^^^
